### PR TITLE
Add 'commit-starts-with' optional parameter

### PR DESCRIPTION
**Pull request by Gudahtt**
An optional parameter has been added to validate that the commit being evaluated starts with a specific string.

If this optional parameter is set, we will check the subject line of the current commit to ensure it matches what was expected. We will replace the string "[version]" in this parameter with the current release version.

The README has been updated to explain how this works.

### CODE REVIEW

**Comment by mcmire** [comment]
I am not sure that a check based on a magic string in a commit message or the name of the release branch would be a sufficient solution to the problem presented in https://github.com/MetaMask/metamask-module-template/pull/153. Anyone, contributors included, could create such a commit and create a PR for it, and we could accidentally merge that PR thinking it was innocuous.

**Comment by Gudahtt** [comment]
It's not a perfect solution. But it seems pretty good for repositories where we're using "Squash & Merge", and creating release branches with `create-release-branch`. The PR title "Release [version]" is kinda hard to miss.

File: README.md
Comment (mcmire) [inline README.md:28]: Should we change this based on how we are using `action-create-release-pr`? Using `startsWith(github.event.commits[0].author.name, 'github-actions')`, that is.

File: action.yml
Comment (mcmire) [inline action.yml]: Nit:

``` suggestion
    description: "Validate that the release commit starts with this string. Use '[version]' to refer to the current release version."
```

File: scripts/is-release.sh
Comment (mcmire) [inline scripts/is-release.sh]: Nit: Should we put the expected format in the message?

``` suggestion
    echo "Notice: commit message does not start with \"${COMMIT_STARTS_WITH}\". Skipping release."
```

**Review by mcmire — COMMENTED** [review]
Looks good, just a few suggestions.

File: README.md
Comment (Gudahtt) [reply README.md:28]: Hmm. I was going to ask about that - `head_commit` does seem to be equivalent to `commits[0]`, except that it's simpler.

We should definitely be using `author.name` over `committer.name`. though. The committer is GitHub for anything merged via GitHub.

File: scripts/is-release.sh
Comment (Gudahtt) [reply scripts/is-release.sh]: Good idea!

File: README.md
Comment (Gudahtt) [reply README.md:28]: I have fixed this in a separate PR: https://github.com/MetaMask/action-is-release/pull/5

**Comment by Gudahtt** [comment]
I can test this with a fork before merging. So far I've tested this just by invoking the Bash script manually in the context of another repo

**Review by rickycodes — DISMISSED** [review]
this all LGTM

File: README.md
Comment (mcmire) [reply README.md:28]: Ah yeah `head_commit` is better. Good catch.

**Review by mcmire — DISMISSED** [review]
LGTM too.

File: action.yml
Comment (Gudahtt) [inline action.yml:24]: Had to push a new commit to quote these arguments. Had a string splitting issue with the commit prefix.

**Review by mcmire — APPROVED** [review]

**Comment by Gudahtt** [comment]
I tested this, and it seems to work. Here is a negative example: https://github.com/Gudahtt/controllers-fork/actions/runs/3518613740/jobs/5897785814
 And here is a positive example: https://github.com/Gudahtt/controllers-fork/actions/runs/3518683757/jobs/5897923800

**Pull request merged**
